packages/cli/src/commands/hydrogen/deploy.ts

```diff
@@ -351,7 +351,9 @@ export async function runDeploy(
     );
   }
 
-  if (!isCI) {
+  if (isCI) {
+    userProvidedEnvironmentTag = envBranch || null;
+  } else {
```

Comment:
Instead of an `else` here I think I'd prefer we bump this to its own conditional:

```js
if (isCi && envBranch) {
  userProvidedEnvironmentTag = envBranch
}

if (!isCi) {
  // existing logic stays the same
}
```